### PR TITLE
Update Versioning.md

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by maririos** [comment]
cc: @JohnTortugo

File: Documentation/Versioning.md [hunk 1]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
-- **PATCH** - Explicit in source, should default to `0`
-- **PRERELEASE** - Explicit in source, should default to `preview1`
-- **REVISION** - Generated based on build scenario
-  - **Local dev builds** - Defaulted to 0
-  - **Official builds** - Supplied generally as part of the conventional, .NET Core specific `OfficialBuildId` or VSTS `Build.BuildNumber` built in parameters.
-- **SHORTDATE** - Generated based on build scenario
-  - **Local dev builds** - Date of current git HEAD
-  - **Official builds** - Supplied generally as part of the conventional, .NET Core specific `OfficialBuildId` or VSTS `Build.BuildNumber` built in parameters.
+
+The format of package versions produced by the build is determined based on `DotNetFinalVersionKind` variable:
```
Comment (maririos) [inline Documentation/Versioning.md]: NIT: "based on the value of the variable `DotNetFinalVersionKind`"

File: Documentation/Versioning.md [hunk 2]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
+
+| DotNetFinalVersionKind   | example                     |
+|--------------------------|-----------------------------|
+| ""                       | "1.2.3-beta.12345.1+abcdef"|
```
Comment (maririos) [inline Documentation/Versioning.md]: Should we add a clarification or note that when the variable doesn't have a value it represents the Dev/Daily  kind?

File: Documentation/Versioning.md [hunk 3]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
+- **MAJOR**, **MINOR**, **PATCH**:
+  Specified in source using `VersionPrefix` .NET Core SDK property, defaults to `1.0.0`.
+  
+- **PRERELEASE**: 
+  Specified in source using `PreReleaseVersionLabel` property.
+  
+- **REVISION**, **SHORTDATE**: 
+  - In official builds the values are derived from build parameter `OfficialBuildId` with format `20yymmdd.r` like so:
+    - REVISION is set to `r` component of `OfficialBuildId`
+    - SHORTDATE is set to `yy` * 1000 + 50 * `mm` + `dd`. In year 2018 the value is in range [`18051`,`18631`].
```
Comment (weshaggard) [inline Documentation/Versioning.md]: It is probably worth adding a note of why the `50*` is interesting.

File: Documentation/Versioning.md [hunk 4]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
-  }
-  ```
-- **SHORTSHA** - Parsed from the current git HEAD
+## Package Version Generation
+
+- **MAJOR**, **MINOR**, **PATCH**:
+  Specified in source using `VersionPrefix` .NET Core SDK property, defaults to `1.0.0`.
+  
+- **PRERELEASE**: 
+  Specified in source using `PreReleaseVersionLabel` property.
```
Comment (weshaggard) [inline Documentation/Versioning.md]: Should all it we default to `ci` for CI builds and `dev` for dev builds.

File: Documentation/Versioning.md (hunk 2 above)
Comment (weshaggard) [reply Documentation/Versioning.md]: Yeah I was also going to point out that this contracts the statement below that there is no revision or shortdates for ci and dev builds.

File: Documentation/Versioning.md [hunk 5]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
+- **REVISION**, **SHORTDATE**: 
+  - In official builds the values are derived from build parameter `OfficialBuildId` with format `20yymmdd.r` like so:
+    - REVISION is set to `r` component of `OfficialBuildId`
+    - SHORTDATE is set to `yy` * 1000 + 50 * `mm` + `dd`. In year 2018 the value is in range [`18051`,`18631`].
+  - In CI and local dev builds REVISION and SHORTDATE are not included in the package version,
+    unless `DotNetUseShippingVersion` is `true`, in which case the values of `yy`, `mm` and `dd` are derived from the current date
+    and `r` = 1.
+
+- **SHORTSHA**:
+  - The first 8 characters for the current git HEAD commit SHA
```
Comment (weshaggard) [inline Documentation/Versioning.md]: Worth calling out that we use `SourceRevisionId` as the property which is passed in or computed if necessary.

File: Documentation/Versioning.md [hunk 6]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
+File version has 4 parts and need to increase every official build. This is especially important when building MSIs. 
+
+```
+MAJOR.MINOR.FILEPATCH.FILEREVISION
+```
+
+- **MAJOR** and **MINOR**: 
+  Specified in `VersionPrefix` variable.   
+- **FILEPATCH**: 
+  Set to PATCH * 100 + `yy`
```
Comment (weshaggard) [inline Documentation/Versioning.md]: We should call out how PATCH is defined. Might also need that for MAJOR/MINOR.

**Comment by maririos** [comment]
FYI @rainersigwald

**Comment by rainersigwald** [comment]
+ @AndyGerlicher

We would like this to not be a rigid requirement for participation in Arcade. Andy set up a meeting to discuss with some folks later this week.

File: Documentation/Versioning.md [hunk 7]
```diff
@@ -63,85 +63,102 @@ Date agnostic versioning is more hassle than it's worth, though having the sha i
...
-    1.0.0
-  ```
\ No newline at end of file
+The repository opts into SemVer1 fallback by setting `SemanticVersioningV1` property to `true`.
+
+Examples of SemVer1 package versions:
+
+| DotNetFinalVersionKind   | example                     |
+|--------------------------|-----------------------------|
+| ""                       | "1.2.3-beta-12345-01-abcdef"|
```
Comment (mmitche) [inline Documentation/Versioning.md]: If this is dev it shouldn't have revision/file.

File: Documentation/Versioning.md (hunk 4 above)
Comment (mmitche) [reply Documentation/Versioning.md]: I like that idea.

File: Documentation/Versioning.md (hunk 2 above)
Comment (tmat) [reply Documentation/Versioning.md]: Done

File: Documentation/Versioning.md (hunk 1 above)
Comment (tmat) [reply Documentation/Versioning.md]: Done

File: Documentation/Versioning.md (hunk 4 above)
Comment (tmat) [reply Documentation/Versioning.md]: Done

File: Documentation/Versioning.md (hunk 6 above)
Comment (tmat) [reply Documentation/Versioning.md]: Done.

**Comment by maririos** [comment]
@tmat it will be good to add the information for `DotNetUseShippingVersions` here as it affects how the version is produced.

**Comment by tmat** [comment]
>  it will be good to add the information for DotNetUseShippingVersions here as it affects how the version is produced.

It's there in "Package Version Generation" section.

**Pull request merged**